biorxiv/word_vector_experiment/biorxiv_visualize_embeddings.py

```diff
@@ -0,0 +1,362 @@
+print(g)
+
+
+# In[15]:
+
+
+g = (
+    p9.ggplot(biorxiv_tsne_models_lastest[250])
+    + p9.aes(x="tsne1", y="tsne2", color="factor(category)")
+    + p9.geom_point()
+    + p9.labs(
+        title="TSNE of BioRxiv (250 dim)",
+        color="Article Category"
+    )
+)
+g.save("output/embedding_output/tsne/figures/biorxiv_tsne_250.png", dpi=500)
+print(g)
+
+
+# In[16]:
+
+
+g = (
+    p9.ggplot(biorxiv_tsne_models_lastest[300])
+    + p9.aes(x="tsne1", y="tsne2", color="factor(category)")
+    + p9.geom_point()
+    + p9.labs(
+        title="TSNE of BioRxiv (300 dim)",
+        color="Article Category"
+    )
+)
+g.save("output/embedding_output/tsne/figures/biorxiv_tsne_300.png", dpi=500)
+print(g)
+
+
+# These plots are similar to each other which results in documents grouping together based on their assigned category. Interestingly, there seems to be a big divide between the neuroscience categories. Plus the bioinformatics papers have a separated blob. Question to follow up on is: why the discrepancies? 
+
+# ## TSNE Outliers
+
+# In[17]:
+
+
+g = (
+    p9.ggplot(biorxiv_tsne_models_lastest[300])
+    + p9.aes(x="tsne1", y="tsne2", color="factor(category)")
+    + p9.geom_point()
+    + p9.annotate("segment", x=60, y=60, xend=45, yend=45, arrow=p9.arrow(), color="red")
+    + p9.labs(
+        title="Outlier Detection of BioRxiv TSNE (300 dim)",
+        color="Article Category"
+    )
+)
+print(g)
+
+
+# In[18]:
+
+
+(
+    biorxiv_tsne_models_lastest[300]
+    .query("tsne1 > 30 & tsne1 < 45")
+    .query("tsne2 > 31 & tsne2 < 45")
+    .query("category != 'bioinformatics'")
+    .category
+    .value_counts()
+)
+
+
+# Looking into the bioinformatics blob approximately 370 articles are off category.
+
+# In[19]:
+
+
+(
+    biorxiv_tsne_models_lastest[300]
+    .query("tsne1 > 30 & tsne1 < 45")
+    .query("tsne2 > 31 & tsne2 < 45")
+    .query("category in ['pharmacology', 'none', 'physiology', 'biochemistry', 'animal behavior and cognition']")
+)
+
+
+# (10.1101/008896) is a bioinformatics and biophysics paper, but the authors didn't provide a category for the preprint. (10.1101/248898) is a preprint that monitors data availability, which suggests that it should have the bioinformatics category as well. (10.1101/377960) should have the bioinformatics category as well.
+
+# In[20]:
+
+
+g = (
+    p9.ggplot(biorxiv_tsne_models_lastest[300])
+    + p9.aes(x="tsne1", y="tsne2", color="factor(category)")
+    + p9.geom_point()
+    + p9.annotate("rect", xmin=-30, xmax=-20, ymin=40, ymax=55, alpha=0.4)
+    + p9.labs(
+        title="Outlier Detection of BioRxiv TSNE (300 dim)",
+        color="Article Category"
+    )
+)
+print(g)
+
+
+# In[21]:
+
+
+(
+    biorxiv_tsne_models_lastest[300]
+    .query("tsne1 > -30 & tsne1 < -20")
+    .query("tsne2 > 40 & tsne2 < 55")
+    .query("category != 'neuroscience'")
+    .category
+    .value_counts()
+)
+
+
+# Looking into the bioinformatics blob approximately 370 articles are off category.
```

Comment:
So these are articles that are currently classified as "not bioinformatics" and your classification is say they should be "bioinformatics"?

Reply:
No classification in this situation just document embeddings. I'm operating under the hypothesis: `documents under the same category should cluster together rather than documents in different categories.`
In layman's terms: blue should cluster with blue instead of blue having random red points. Since that isn't the case, I needed to look more into why, hence the output tables in the notebook.